Kudu.Core/Functions/FunctionManager.cs

```diff
@@ -422,5 +425,113 @@ private string GetFunctionSecretsFilePath(string functionName)
         {
             return Path.Combine(_environment.DataPath, Constants.Functions, Constants.Secrets, $"{functionName}.json");
         }
+
+        /// <summary>
+        /// Populates a <see cref="ZipArchive"/> with the content of the function app.
+        /// It can also include local.settings.json and .csproj files for a full Visual Studio project.
+        /// sln file is not included since it changes between VS versions and VS can auto-generate it from the csproj.
+        /// All existing functions are added as content with "Always" copy to output.
+        /// </summary>
+        /// <param name="zip">the <see cref="ZipArchive"/> to be populated with function app content.</param>
+        /// <param name="includeAppSettings">Optional: indicates whether to add local.settings.json or not to the archive. Default is false.</param>
+        /// <param name="includeCsproj">Optional: indicates whether to add a .csproj to the archive. Default is false.</param>
+        /// <param name="projectName">Optional: the name for *.csproj file if <paramref name="includeCsproj"/> is true. Default is appName.</param>
+        /// <returns></returns>
+        public IList<ZipArchiveEntry> CreateArchive(ZipArchive zip, bool includeAppSettings = false, bool includeCsproj = false, string projectName = null)
```

Comment:
Not sure I understand this code. This returns a `IList<ZipArchiveEntry>`, but I don't see any code actually using that.